test/Shared/UserManagerTestBase.cs

```diff
@@ -83,15 +83,10 @@ protected RoleManager<TRole> CreateRoleManager(object context = null, IServiceCo
         protected abstract void AddUserStore(IServiceCollection services, object context = null);
         protected abstract void AddRoleStore(IServiceCollection services, object context = null);
 
-        protected TUser CreateTestUser(string namePrefix = "")
-        {
-            return new TUser() { UserName = namePrefix + Guid.NewGuid().ToString() };
-        }
+        protected abstract TUser CreateTestUser(string username = "", string email = "", string phoneNumber = "",
+            bool lockoutEnabled = false, DateTimeOffset? lockoutEnd = null);
```

Comment:
This is test code not user code, you are creating a pit of failure for us, as if we write two tests that happen to choose the same user name now, they will fail/collide/ do bad things since the user name will exist.